### PR TITLE
GRIDEDIT-1491: Add Mesh2dGetNodeEdgeData 

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -6,7 +6,7 @@
 		<AssemblyTitle>$(MSBuildProjectName)</AssemblyTitle>
 
 		<BuildNumber>0</BuildNumber>
-		<MeshKernelNETVersion>5.0.1</MeshKernelNETVersion>
+		<MeshKernelNETVersion>5.0.2</MeshKernelNETVersion>
```
Comment (andreasbuykx) [inline Directory.Build.props:9]: updating from 5.0.1 to 5.0.2 should not be part of this PR

File: src/MeshKernelNET/Api/IMeshKernelApi.cs [hunk 2]
```diff
@@ -1075,14 +1075,23 @@ int Mesh2dGetClosestNode(int meshKernelId, double xCoordinateIn, double yCoordin
 
 
         /// <summary>
-        /// Gets the grid state as a <see cref="Mesh2DNative"/> structure including the cell information
+        /// Gets the grid state as a <see cref="DisposableMesh2D"/> structure including the cell information
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Api/IMeshKernelApi.cs:1078]: Document that vertex coordinates, edge connectivity, face connectivity, edge midpoints and face center points are initialized.

File: src/MeshKernelNET/Api/IMeshKernelApi.cs [hunk 3]
```diff
@@ -1075,14 +1075,23 @@ int Mesh2dGetClosestNode(int meshKernelId, double xCoordinateIn, double yCoordin
...
         /// <param name="meshKernelId">Id of the grid state</param>
         /// <param name="disposableMesh2D">
         /// <see cref="DisposableMesh2D"/> with the grid state
         /// <returns>Error code</returns>
         int Mesh2dGetData(int meshKernelId, out DisposableMesh2D disposableMesh2D);
 
+        /// <summary>
+        /// Gets the mesh state as a <see cref="DisposableMesh2D"/> structure, only the node and edge data
+        /// </summary>
+        /// <param name="meshKernelId">Id of the grid state</param>
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Api/IMeshKernelApi.cs:1089]: Document that only vertex coordinates and edge connectivity are initialized.

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs [hunk 4]
```diff
@@ -3454,6 +3454,52 @@ public void Mesh2dExpungeStateThroughApi()
                 }
             }
         }
+
+        [Test]
+        public void Mesh2dGetNodeEdgeDataThroughApi()
+        {
+
+            // Setup
+            using (DisposableMesh2D mesh = CreateMesh2D(10, 10, 10, 10))
```
Comment (andreasbuykx) [inline test/MeshKernelNETTest/Api/MeshKernelTest.cs:3463]: use different numbers for input argument values to be more likely to hit errors like swapped arguments etc.

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs [hunk 5]
```diff
@@ -3454,6 +3454,52 @@ public void Mesh2dExpungeStateThroughApi()
...
+        [Test]
+        public void Mesh2dGetNodeEdgeDataThroughApi()
+        {
+
+            // Setup
+            using (DisposableMesh2D mesh = CreateMesh2D(10, 10, 10, 10))
+            using (var api = new MeshKernelApi())
+            {
+                var id = 0;
+                var meshRetrived = new DisposableMesh2D();
```
Comment (andreasbuykx) [inline test/MeshKernelNETTest/Api/MeshKernelTest.cs:3467]: typo: meshRetrieved

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs [hunk 6]
```diff
@@ -3454,6 +3454,52 @@ public void Mesh2dExpungeStateThroughApi()
...
+                try
```
Comment (andreasbuykx) [inline test/MeshKernelNETTest/Api/MeshKernelTest.cs:3468]: I thought the api doesn't throw? Or is it MKNet that deals with that?

**Review by andreasbuykx — APPROVED** [review]

File: test/MeshKernelNETTest/Api/MeshKernelTest.cs (hunk 6 above)
Comment (lucacarniato) [reply test/MeshKernelNETTest/Api/MeshKernelTest.cs:3468]: Most MKNet API implementation is composed of calls to mk, where exceptions are handled and codes are returned. We can remove most of the try/finally blocks, but it needs to be consistent across all tests
